### PR TITLE
Task/141 update network wallet balance

### PR DESCRIPTION
**Pull request by storkandstars**
# Overview

Updating balance of a user's NetworkWallet. Wallet Repository exposes methods to refresh balances for AccountWallet, AssetWallet and NetworkWallet. Account update lifetime is 180s by default.

## Test cases

1. Add an asset to inventory
2. Go to receive screen
3. Send resources to the wallet
4. Go to Inventory. Balance on network wallet should be updated. Remember balance is updated every 3 minutes. So enter to inventory after 3 minutes passed.

### CODE REVIEW

File: lib/data/repositories/wallet_repository.dart [hunk 1]
```diff
@@ -464,6 +470,156 @@ class WalletRepository {
...
+
+  Future<AccountWallet> refreshNetworkWalletBalance(
+    AccountWallet accountWallet, {
+    required NetworkWallet networkWallet,
+    bool save = true,
+  }) async {
+    try {
+      final netWallet = await _updateNetworkWalletBalance(networkWallet);
+      final assWallet = accountWallet.wallets.firstWhere(
+          (w) => w.findWallet(networkWallet.uuid) != null ? true : false);
```
Comment (jvrastil) [inline lib/data/repositories/wallet_repository.dart:516]: can it be shorter?
```
firstWhere((w) => w.findWallet(networkWallet.uuid) != null
```

Comment (storkandstars) [reply lib/data/repositories/wallet_repository.dart]: Fixed here https://github.com/SimplioOfficial/simplio-app/pull/217/commits/7c5261cf3d3544c5024d4b3802681e4394e00a5b

**Comment by jvrastil** [comment]
Also, shouldn't we enable user to refresh the balance so he does not have to wait 3 mins to display updated balance...?

(no need to do it in this task though)

**Comment by storkandstars** [comment]
Yes. It is planned for the next sprint.

**Review by jvrastil — APPROVED** [review]

**Pull request merged**